### PR TITLE
fix: tolerate floating-point residue at pad boundaries

### PR DESCRIPTION
**Pull request by AnasSarkiz**
## Summary

- accept only sub-nanometer floating-point residue when a trace endpoint lands mathematically on a pad boundary
- apply the numerical tolerance consistently to supported SMT-pad and plated-hole boundary shapes
- keep real physical gaps outside the pad rejected
- cover the exact reduced dataset01 circuit018 geometry

## Root cause

Dataset01 `circuit018` computes the preloaded trace endpoint and rectangular pad boundary through separate decimal calculations. Their mathematically equal coordinates differ by about `6e-16 mm`, so a strict comparison reports a false missing-pad connection.

This fixes the shared containment predicate in `@tscircuit/checks`. It does **not** move or normalize the trace endpoint, alter evaluator geometry, or special-case circuit018.

## Before / after

The stack merged while the snapshot-body update was being prepared. The visuals below are pinned to [upstream snapshot commit `719e845`](https://github.com/tscircuit/checks/commit/719e8451111f6eeb011c80952964b48d14a9d375), which adds only the requested paired visual artifacts around the same merged circuit018 regression.

| Before: false disconnect | After: valid boundary contact |
| --- | --- |
| circuit018 before | circuit018 after |

The before snapshot renders the actual error reported by `main` for this fixture. The after snapshot preserves the same copper geometry with no DRC overlay.

## Fix

`isPointInPad` allows `1e-9 mm` of numerical residue at pad boundaries. The regression separately checks that a point `1e-6 mm` outside remains rejected. Rounded SMT-pad and circular plated-hole boundary cases are covered as well.

## Validation

- `bun test` — 177 passed, 0 failed
- `bun run build`
- focused Biome formatting check
- `git diff --check`

### CODE REVIEW

File: tests/lib/__snapshots__/circuit018-pad-boundary-before.snap.svg
Comment (ShiboSoftwareDev) [inline tests/lib/__snapshots__/circuit018-pad-boundary-before.snap.svg:1]: this is not the correct way of showing old behavior, a good way is to have a stack, one with the repro and another with the fix so we'd see the diff

File: lib/check-traces-are-contiguous/is-point-in-pad.ts [hunk 1]
```diff
@@ -78,7 +86,8 @@ export function isPointInPad(
       if (pad.shape === "rotated_pill") {
         const pill = getPillCenterLineForPad(pad)
         return (
-          pointToSegmentDistance(point, pill.start, pill.end) <= pill.radius
+          pointToSegmentDistance(point, pill.start, pill.end) <=
+          pill.radius + POINT_IN_PAD_TOLERANCE_MM
```
Comment (ShiboSoftwareDev) [inline lib/check-traces-are-contiguous/is-point-in-pad.ts:90]: you added the logic for a lot of shapes and tested only one, maybe add a couple more tests

**Review by ShiboSoftwareDev — APPROVED** [review]

**Review by rushabhcodes — APPROVED** [review]
